src/overview/onboarding/notification-helper.ts

```diff
@@ -0,0 +1,134 @@
+ * "browse around a bit" in Power Search welcome notification.
+ */
+
+const handler = toolbarNotifications => async () => {
+    toolbarNotifications._destroyRootElement()
+    toolbarNotifications.showToolbarNotification('go-to-dashboard')
+
+    processEventRPC({
+        type: EVENT_NAMES.POWERSEARCH_BROWSE_PAGE,
+    })
+
+    await utils.setOnboardingStage(
+        'powerSearch',
+        STAGES.powerSearch.overviewTooltips,
+    )
+}
+
+/**
+ * Shows Toolbar notifications on website based on
+ * onboarding flags set in local storage.
+ * @param toolbarNotifications ToolbarNotification instance to trigger notification
+ */
+export const conditionallyShowOnboardingNotifications = async ({
+    toolbarNotifications,
+}) => {
+    /*
+    Fetch shouldShowOnboarding and return if it's false as 
+    that would mean the user has closed the onboarding demo.
+    */
+    const shouldShowOnboarding = await getLocalStorage(
+        STORAGE_KEYS.shouldShowOnboarding,
+        true,
+    )
+    if (!utils.isDemoPage && !shouldShowOnboarding) {
+        return
+    }
+
+    const {
+        annotationStage,
+        powerSearchStage,
+        taggingStage,
+    } = await utils.fetchAllStages()
+
+    if (annotationStage === STAGES.redirected) {
+        toolbarNotifications.showToolbarNotification('onboarding-higlight-text')
+        await utils.setOnboardingStage(
+            'annotation',
+            STAGES.annotation.notifiedHighlightText,
+        )
+    }
+
+    if (powerSearchStage === STAGES.redirected) {
+        const position = utils.getPageCenter()
+        toolbarNotifications._destroyRootElement()
+        toolbarNotifications.showToolbarNotification('power-search-browse', {
+            position,
+            triggerNextNotification: handler(toolbarNotifications),
+        })
+        await utils.setOnboardingStage(
+            'powerSearch',
+            STAGES.powerSearch.notifiedBrowsePage,
+        )
+    }
+
+    if (taggingStage === STAGES.redirected) {
+        toolbarNotifications.showToolbarNotification('tag-this-page')
+        await utils.setOnboardingStage(
+            'tagging',
+            STAGES.tagging.notifiedTagPage,
+        )
+    }
+}
+
+/**
+ * Conditionally removes the Select Option notifcation in Annotation
+ * Onboarding Flow. Either used when user clicks outside or an annotation
+ * is created.
+ * @param nextStage Next stage to set for annotations flow
+ */
+export const conditionallyRemoveSelectOption = async nextStage => {
+    const annotationStage = await utils.fetchOnboardingStage('annotation')
+    console.log(annotationStage)
+    if (annotationStage === STAGES.annotation.notifiedSelectOption) {
+        await utils.setOnboardingStage('annotation', nextStage)
+        // Close the curren select-option notification manually since
+        // accessing the toolbarNotification instance from here is not possible
+        document.querySelector('.memex-tooltip-notification').remove()
```

Comment:
This seems to be a repeat of what's already in the `destroyRootElement` export from `src/toolbar-notification/content_script/rendering` module. Is it appropriate to use that here? (note that module seems to be content script, while this is main UI script)

Reply:
I didn't notice that export. These exports are imported in `content-tooltip`. Moved the code here from there because most of the code is onboarding logic.